deepchecks/tabular/context.py

```diff
@@ -221,8 +221,16 @@ def __init__(
                 template='For more information please refer to the Supported Models guide {link}')
             raise DeepchecksValueError(f'Received unsorted model_classes. {supported_models_link}')
 
-        self._task_type, self._observed_classes, self._model_classes = infer_task_type_and_classes(
-            model, train, test, model_classes)
+
+        observed_labels, self._model_classes = get_labels_and_classes(
+            model, train, test, y_pred_train, y_pred_test, model_classes
+        )
+
+        self._task_type = infer_task_type(train, observed_labels, self._model_classes)
+        if self._task_type in (TaskType.BINARY, TaskType.MULTICLASS):
+            self._observed_classes = sorted(observed_labels.dropna().unique().tolist())
+        else:
+            self._observed_classes = None
```

Comment:
Why repeat this and not just use infer_task_type_and_classes?

Reply:
infer task type and classes is just a shortcut for other places, it gets only model and dataset

Reply:
umm never mind, it's possible